backend/cn/lib/solver.ml

```diff
@@ -1238,17 +1238,26 @@ let make globals =
   s
 
 
+let model_evaluator_solver = ref None
+
```

Comment:
Unless this needs to be shared/used by other functions, then something like this is preferable.
```ocaml
(** Evaluate terms in the context of a model computed by the solver. *)
let model_evaluator =
  let let model_evaluator_solver = ref None in
  fun solver mo ->
  match SMT.to_list mo with
```